pytapo/__init__.py

```diff
@@ -1086,6 +1086,80 @@ def getWhitelampStatus(self):
             "getWhitelampStatus", {"image": {"get_wtl_status": ["null"]}}
         )
 
+    ### TODO: UNTESTED
+    def getFloodlightStatus(self):
+        return self.executeFunction(
+            "getFloodlightStatus", {"floodlight": {"get_floodlight_status": ""}}
+        )
+
+    ### TODO: UNTESTED
+    def setFloodlightStatus(self, status):
+        return self.executeFunction(
+            "setFloodlightStatus",
+            {"floodlight": {"set_floodlight_status": 1 if status else 0}},
+        )
+
+    ### TODO: UNTESTED
+    def getFloodlightConfig(self):
+        return self.executeFunction(
+            "getFloodlightConfig", {"floodlight": {"name": "config"}}
+        )
+
+    ### TODO: UNTESTED
+    def setFloodlightConfig(
+        self,
+        autoOffEnabled: bool,
+        scheduleMode,
+        endTime,
+        imgDetTriEnabled: bool,
+        intensityLevel: int,
+        scheduleEnabled: bool,
+        manualDuration: int,
+        startTime: int,
+        sunriseOffset: int,
+        sunsetOffset: int,
+        triggerDuration: int,
+    ):
+        ### TODO: Can we set only one value, not all?
+        return self.executeFunction(
+            "setFloodlightConfig",
+            {
+                "floodlight": {
+                    "config": {
+                        "schedule_mode": scheduleMode,
+                        "auto_off_enabled": "on" if autoOffEnabled else "off",
+                        "end_time": endTime,
+                        "img_det_tri_enabled": "on" if imgDetTriEnabled else "off",
+                        "intensity_level": str(intensityLevel),
+                        "schedule_enabled": "on" if scheduleEnabled else "off",
+                        "manual_duration": str(manualDuration),
+                        "start_time": str(startTime),
+                        "sunrise_offset": str(sunriseOffset),
+                        "sunset_offset": str(sunsetOffset),
+                        "trigger_duration": str(triggerDuration),
+                    }
+                }
+            },
+        )
+
+    ### TODO: UNTESTED
+    def getFloodlightCapability(self):
+        return self.executeFunction(
+            "getFloodlightCapability", {"floodlight": {"name": "capability"}}
+        )
```

Comment:
If it works, we need to probably extract some specific property, just like in other functions